src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimAnalysisVisitor.cs

```diff
@@ -51,14 +51,17 @@ public override MultiValue VisitConversion (IConversionOperation operation, Stat
 
 		public override MultiValue VisitParameterReference (IParameterReferenceOperation paramRef, StateValue state)
 		{
-			return new MethodParameterValue (paramRef.Parameter);
+			return paramRef.Parameter.Type.IsTypeInterestingForDataflow () ? new MethodParameterValue (paramRef.Parameter) : TopValue;
 		}
 
 		public override MultiValue VisitInstanceReference (IInstanceReferenceOperation instanceRef, StateValue state)
 		{
 			if (instanceRef.ReferenceKind != InstanceReferenceKind.ContainingTypeInstance)
 				return TopValue;
 
+			if (instanceRef.Type != null && !instanceRef.Type.IsTypeInterestingForDataflow ())
+				return TopValue;
```

Comment:
We should return TopValue if the instance ref doesn't have a type as well - the condition is technically the other way round "Return tracked value only if the type is interesting" - null types are by definition "not interesting".